### PR TITLE
Debug server & tiered distribution

### PR DESCRIPTION
**Pull request by shubhitms**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -28,34 +29,43 @@ type SidecarOptions = {
...
   repositoryOwner: string
   repositoryName: string
   updateIntervalMs?: number
+  transportProtocol?: TransportProtocol
+  serverPort?: number,
 }
 
 const defaultUpdateIntervalMs = 30 * 1000; // 30s
 
-async function initBackendInMemoryClient(options: BackendOptions): Promise<Client> {
```
Comment (shubhitms) [inline src/index.ts:52]: change mode names to match our updated spec

File: src/index.ts [hunk 2]
```diff
@@ -28,34 +29,43 @@ type SidecarOptions = {
...
+async function initCachedAPIClient(options: BackendOptions): Promise<Client> {
   const transport = await new ClientTransportBuilder({
-    hostname: options.hostname || "https://prod.api.lekko.dev",
-    protocol: TransportProtocol.HTTP,
+    hostname: options.hostname ?? "https://prod.api.lekko.dev",
+    protocol: options.transportProtocol ?? TransportProtocol.HTTP,
```
Comment (shubhitms) [inline src/index.ts:59]: allow overriding the default url and protocol to support tiered distribution. you can now run the cached API provider with the sidecar URL and gRPC protocol for tiered distro

File: src/memory/store.ts [hunk 3]
```diff
@@ -85,8 +88,48 @@ export class Store {
...
+    contents.namespaces.sort((a, b) => {
+        return a.name.localeCompare(b.name);
+    });
+    return contents;
+}
+
+export function hashContents(contents: GetRepositoryContentsResponse): string {
+    return createHash('sha256')
+        .update(contents.toBinary())
+        .digest().toString('hex');
```
Comment (shubhitms) [inline src/memory/store.ts:133]: fix to make sure the hash encoding matches the go-sdk

File: src/transport-builder.ts [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
   HTTP,
   gRPC,
 }
 
-const APIKEY_INTERCEPTOR = (apiKey: string) => (next: any) => async (req: any) => {
-  req.header.set('apikey', apiKey);
+const APIKEY_INTERCEPTOR = (apiKey?: string) => (next: any) => async (req: any) => {
+  if (apiKey) {
+    req.header.set('apikey', apiKey);
+  }
```
Comment (shubhitms) [inline src/transport-builder.ts:12]: only add the header if apikey exists, to support tiered distro

File: src/transport-builder.ts [hunk 5]
```diff
@@ -27,17 +29,18 @@ export class ClientTransportBuilder {
       if (this.apiKey === undefined) {
         throw new Error("API Key required");
       }
-      return await createConnectTransport({
+      return createConnectTransport({
```
Comment (passichenko) [inline src/transport-builder.ts:32]: Why this change?

**Review by passichenko — APPROVED** [review]

File: src/transport-builder.ts (hunk 5 above)
Comment (shubhitms) [reply src/transport-builder.ts:32]: its not an async function, so there was no use to call await
